compliance_suite/validate_response.py

```diff
@@ -26,54 +26,58 @@ def set_case(self, case):
         self.case = case
 
     def get_schema(self, schema_file_name):
-        """This function loads the given schema available"""
+        """
+        Loads the given schema if available,
+        Throws an error if the file does not exist
+        """
+        if not os.path.exists(schema_file_name):
+            raise ValueError(f"Schema file '{schema_file_name}' does not exist")
         with open(schema_file_name, 'r') as file:
             schema = json.load(file)
-        # TODO: throw exception -> fail case if schema file does not exist or add unittest to make sure all the schema files are available
         return schema
 
     def validate_response_schema(self):
-        if self.case.get_status() != Status.SKIP:
-            expected_schema_file_path = SCHEMA_DIR + "/" + self.response_schema_file
-            expected_schema = self.get_schema(expected_schema_file_path)
-            absolute_schema_file_path = os.path.dirname(os.path.abspath(expected_schema_file_path))
-            reference_resolver = jsonschema.RefResolver(base_uri = "file://"+absolute_schema_file_path+"/", referrer = None)
-            try:
-                validate(instance = self.actual_response.json(),resolver = reference_resolver,schema=expected_schema)
-                self.case.set_message("Schema Validation Successful")
-                self.case.set_status_pass()
-            except Exception as e:
-                self.case.set_message(e.message if hasattr(e,"message") else str(e))
-                self.case.add_log_message("Stack Trace: "+str(e))
-                self.case.set_status_fail()
-        else:
-            pass
-            # make sure if skip, the reason is added to message and/ log_message
```

Comment:
I don't think the log was implemented. Should we keep the comment for future?

Reply:
There are no logs implemented. However we do have a logging class, which makes it easier to implement logs. I have added a ticket for logging - [GA4GHTT90](https://www.ebi.ac.uk/panda/jira/browse/GA4GHTT-90)